tests/unit/test_transforms.py

```diff
@@ -1143,3 +1146,55 @@ def test_crop_covered_segments_can_set_fully_covered_segment_behavior(
                 Dataset(actual).init_cache()
 
             assert "completely covered object removed" in str(capture.value)
+
+
+TRANSFORMS = [
+    transform_cls
+    for transform_cls in DEFAULT_ENVIRONMENT.transforms.items.values()
+    if transform_cls.__module__ == "datumaro.plugins.transforms"
+]
+
+
+@pytest.mark.parametrize("transform_cls", TRANSFORMS)
+def test_transform_fields(transform_cls):
+    if transform_cls is transforms.RemoveItems:
+        pytest.skip()
+
+    modified_fields = set()
+
+    class _DatasetItem(DatasetItem):
+        def wrap(self, **kwargs):
+            if "annotations" in kwargs and transform_cls is not transforms.RemoveAnnotations:
+                assert callable(kwargs["annotations"])
+            modified_fields.update(kwargs.keys())
+
+            return super().wrap(**kwargs)
+
+    source_dataset = Dataset.from_iterable(
+        [
+            _DatasetItem(
+                id=f"item_{index}",
+                subset="subset",
+                media=Image.from_file(path=f"image_path_{index}.jpg", size=(100, 100)),
+                annotations=[],
+            )
+            for index in range(10)
+        ],
+        categories=["aaa", "bbbb"],
+    )
+
+    parameters = {
+        "UpdateInfos": dict(dst_infos={}),
+        "Rename": dict(regex="|item|foo|"),
+        "RemapLabels": dict(mapping={}),
+        "ProjectLabels": dict(dst_labels={}),
+        "ResizeTransform": dict(width=10, height=10),
+        "RandomSplit": dict(splits=[("train", 0.67), ("test", 0.33)]),
+    }.get(transform_cls.__name__, {})
+
+    dataset = source_dataset.transform(transform_cls, **parameters)
+
+    assert len(list(dataset)) == 10
+
+    subsets_modified = bool({"subset"} & modified_fields)
```

Comment:
```suggestion
    subsets_modified = "subset" in modified_fields
```

Although honestly, since this appears to be the only use of `modified_fields`, you might as well just have `_DatasetItem` set `subsets_modified` directly.

Reply:
done